commit 916b4eaefa2cbd09defbd57b1f89b3fcced4c9fe
Author: Jeroen Demeyer <jdemeyer@cage.ugent.be>
Date:   Thu Jan 25 10:25:15 2018 +0100

    Replace mpc_set_fr_fr by two mpfr_set calls

commit 82cfcac492b69ce3741d036e1741f58896ad1e48
Author: Jeroen Demeyer <jdemeyer@cage.ugent.be>
Date:   Thu Jan 25 10:23:49 2018 +0100

    Add data structures for the various types

diff --git a/src/gmpy2.pxd b/src/gmpy2.pxd
index 7cdcc72..6d7d092 100644
--- a/src/gmpy2.pxd
+++ b/src/gmpy2.pxd
@@ -1,14 +1,21 @@
 cdef extern from "gmp.h":
     # gmp integers
+    ctypedef long mp_limb_t
+
     ctypedef struct __mpz_struct:
-        pass
+        int _mp_alloc
+        int _mp_size
+        mp_limb_t* _mp_d
+
     ctypedef __mpz_struct mpz_t[1]
     ctypedef __mpz_struct *mpz_ptr
     ctypedef const __mpz_struct *mpz_srcptr

     # gmp rationals
     ctypedef struct __mpq_struct:
-        pass
+        __mpz_struct _mp_num
+        __mpz_struct _mp_den
+
     ctypedef __mpq_struct mpq_t[1]
     ctypedef __mpq_struct *mpq_ptr
     ctypedef const __mpq_struct *mpq_srcptr
@@ -21,8 +28,16 @@ cdef extern from "gmp.h":

 cdef extern from "mpfr.h":
     # mpfr reals
+    ctypedef int mpfr_sign_t
+    ctypedef long mpfr_prec_t
+    ctypedef long mpfr_exp_t
+
     ctypedef struct __mpfr_struct:
-        pass
+        mpfr_prec_t _mpfr_prec
+        mpfr_sign_t _mpfr_sign
+        mpfr_exp_t _mpfr_exp
+        mp_limb_t* _mpfr_d
+
     ctypedef __mpfr_struct mpfr_t[1]
     ctypedef __mpfr_struct *mpfr_ptr
     ctypedef const __mpfr_struct *mpfr_srcptr
@@ -45,7 +60,9 @@ cdef extern from "mpfr.h":
 cdef extern from "mpc.h":
     # mpc complexes
     ctypedef struct __mpc_struct:
-        pass
+        mpfr_t re
+        mpfr_t im
+
     ctypedef __mpc_struct mpc_t[1];
     ctypedef __mpc_struct *mpc_ptr;
     ctypedef const __mpc_struct *mpc_srcptr;
@@ -147,5 +164,9 @@ cdef inline mpc GMPy_MPC_From_mpc(mpc_srcptr c):
 # Build a gmpy2 mpc from a real part mpfr and an imaginary part mpfr
 cdef inline mpc GMPy_MPC_From_mpfr(mpfr_srcptr re, mpfr_srcptr im):
     cdef mpc res = GMPy_MPC_New(mpfr_get_prec(re), mpfr_get_prec(im), NULL)
-    mpc_set_fr_fr(res.c, re, im, MPC_RNDNN)
+    # We intentionally use MPFR funtions instead of MPC functions here
+    # in order not to add an unneeded dependency on MPC. It's probably
+    # faster too this way.
+    mpfr_set(res.c.re, re, MPFR_RNDN)
+    mpfr_set(res.c.im, im, MPFR_RNDN)
     return res
